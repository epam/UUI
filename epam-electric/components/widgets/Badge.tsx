import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';

type BadgeMods = {
    /**
     * Defines component color.
     * @default 'info'
     */
    color?: uui.BadgeProps['color'] | 'yellow'| 'orange' | 'purple' | 'cyan' | 'mint' | 'cobalt' | 'violet' | 'fuchsia' | 'white' | 'night100' | 'night600';
    /**
     * Defines component fill style.
     * @default 'solid'
     */
    fill?: uui.BadgeProps['fill'];
    /**
     * Defines component size.
     * @default '36'
     */
    size?: uui.BadgeProps['size'];
};

/** Represents the properties of a Badge component. */
export type BadgeProps = uui.BadgeCoreProps & BadgeMods;

export const Badge = /* @__PURE__ */createSkinComponent<uui.BadgeProps, BadgeProps>(
    uui.Badge,
    (props) => ({
        color: props.color || 'info',
    }),
);
