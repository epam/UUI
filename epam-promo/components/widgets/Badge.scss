@use  '../../assets/styles/index' as *;

@mixin transparent-button-layout() {
    padding-left: 0;
    padding-right: 0;
}

.root {
    :global(.uui-count) {
        font-family: $font-sans-semibold;
        margin: auto;
        height: 18px;
        line-height: 18px;
        font-size: 12px;
        border-radius: 9px;
        padding-left: 6px;
        padding-right: 6px;
        min-width: 6px;
        text-align: center;
    }

    :global(.uui-count):not(:first-child) {
        margin-left: 6px;
    }

    :global(.uui-caption) {
        font-family: $font-sans;
    }

    &:global(.uui-button-box) {
        border-width: 1px;

    }

    &.size-18 {
        min-width: 18px;
        min-height: 18px;
        padding-left: 3px;
        padding-right: 3px;
        border-radius: 9px;

        :global(.uui-count) {
            height: 12px;
            line-height: 12px;
            font-size: 8px;
            border-radius: 6px;
            padding-left: 3px;
            padding-right: 3px;
        }

        :global(.uui-count):not(:first-child) {
            margin-left: 3px;
        }

        :global(.uui-caption) {
            padding: 2px 3px;
            line-height: 12px;
            font-size: 12px;
        }

        &.fill-transparent {
            @include transparent-button-layout;

            :global(.uui-caption) {
                margin-left: 17px;
                padding-right: 0;
                padding-left: 0;
            }

            &::before {
                width: 8px;
                height: 8px;
                top: 4px;
                left: 1px;
            }
        }
    }

    &.size-24 {
        @include button-horizontal-layout(24px, 1px);
        border-radius: 12px;
        min-height: 24px;

        :global(.uui-caption) {
            padding-top: 2px;
            padding-bottom: 2px;
            line-height: 18px;
            font-size: 14px;
        }

        &.fill-transparent {
            @include transparent-button-layout;

            :global(.uui-caption) {
                margin-left: 23px;
                padding-right: 0;
                padding-left: 0;
            }

            &::before {
                width: 12px;
                height: 12px;
                top: 5px;
                left: 2px;
            }
        }
    }

    &.size-30 {
        @include button-horizontal-layout(30px, 1px);
        border-radius: 15px;
        min-height: 30px;

        :global(.uui-caption) {
            padding-top: 5px;
            padding-bottom: 5px;
            line-height: 18px;
            font-size: 14px;
        }

        &.fill-transparent {
            @include transparent-button-layout;

            :global(.uui-caption) {
                margin-left: 23px;
                padding-right: 0;
                padding-left: 0;
            }

            &::before {
                width: 12px;
                height: 12px;
                top: 8px;
                left: 2px;
            }
        }
    }

    &.size-36 {
        @include button-horizontal-layout(36px, 1px);
        border-radius: 18px;
        min-height: 36px;

        :global(.uui-caption) {
            padding-top: 5px;
            padding-bottom: 5px;
            line-height: 24px;
            font-size: 14px;
        }

        &.fill-transparent {
            @include transparent-button-layout;

            :global(.uui-caption) {
                margin-left: 23px;
                padding-right: 0;
                padding-left: 0;
            }

            &::before {
                width: 12px;
                height: 12px;
                top: 11px;
                left: 2px;
            }
        }
    }

    &.size-42 {
        @include button-horizontal-layout(42px, 1px);
        border-radius: 21px;
        min-height: 42px;

        :global(.uui-count) {
            height: 24px;
            line-height: 24px;
            font-size: 14px;
            border-radius: 12px;
            min-width: 12px;
        }

        :global(.uui-caption) {
            padding-top: 8px;
            padding-bottom: 8px;
            line-height: 24px;
            font-size: 16px;
        }

        &.fill-transparent {
            @include transparent-button-layout;

            :global(.uui-caption) {
                margin-left: 29px;
                padding-right: 0;
                padding-left: 0;
            }

            &::before {
                width: 16px;
                height: 16px;
                top: 12px;
                left: 3px;
            }
        }
    }

    &.size-48 {
        @include button-horizontal-layout(48px, 1px);
        border-radius: 24px;
        min-height: 48px;

        :global(.uui-count) {
            height: 24px;
            line-height: 24px;
            font-size: 14px;
            border-radius: 12px;
            min-width: 12px;
        }

        :global(.uui-caption) {
            padding-top: 11px;
            padding-bottom: 11px;
            line-height: 24px;
            font-size: 16px;
        }

        &.fill-transparent {
            @include transparent-button-layout;

            :global(.uui-caption) {
                margin-left: 29px;
                padding-right: 0;
                padding-left: 0;
            }

            &::before {
                width: 16px;
                height: 16px;
                top: 15px;
                left: 3px;
            }
        }
    }

    &.fill-solid {
        background-color: var(--background-color);
        color: var(--uui-caption-solid);
        fill: var(--uui-caption-solid);
        border-color: var(--background-color);

        :global(.uui-count) {
            background-color: var(--count-background-color);
            color: var(--count-color);
        }

        &:global(.-clickable) {
            &:hover {
                background-color: var(--background-color-hover);
                border-color: var(--background-color-hover);
                cursor: pointer;
            }
        }
    }

    &.fill-semitransparent {
        background-color: var(--background-color-semitransparent);
        color: var(--uui-caption-semitransparent);
        fill: var(--uui-caption-semitransparent);
        border-color: var(--border-color-semitransparent);

        :global(.uui-count) {
            background-color: var(--count-background-color);
            color: var(--count-color);
        }

        &:global(.-clickable) {
            &:hover {
                background-color: var(--background-color-semitransparent-hover);
                color: var(--uui-caption-semitransparent-hover);
                fill: var(--uui-caption-semitransparent-hover);
                cursor: pointer;
            }
        }
    }

    &.fill-transparent {
        background-color: transparent;
        color: var(--uui-caption-transparent);
        fill: var(--uui-caption-transparent);
        border-color: transparent;

        &:global(.-clickable) {
            &:hover {
                fill: var(--background-color-hover);
                cursor: pointer;
            }
        }

        &::before {
            position: absolute;
            content: '';
            border-radius: 50%;
            background-color: var(--background-color);
        }
    }
}
