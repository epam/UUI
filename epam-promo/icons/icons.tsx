import { ReactComponent as BtnCross12 } from './btn-cross-12.svg';
import { ReactComponent as BtnCross18 } from './btn-cross-18.svg';
import { ReactComponent as BtnCross24 } from './btn-cross-24.svg';
import { ReactComponent as FoldingArrow12 } from './folding-arrow-12.svg';
import { ReactComponent as FoldingArrow18 } from './folding-arrow-18.svg';
import { ReactComponent as FoldingArrow24 } from './folding-arrow-24.svg';
import { ReactComponent as Accept12 } from './accept-12.svg';
import { ReactComponent as Accept18 } from './accept-18.svg';
import { ReactComponent as Accept24 } from './accept-24.svg';
import { ReactComponent as Search12 } from './search-12.svg';
import { ReactComponent as Search18 } from './search-18.svg';
import { ReactComponent as Search24 } from './search-24.svg';
import { ReactComponent as Calendar12 } from './calendar-12.svg';
import { ReactComponent as Calendar18 } from './calendar-18.svg';
import { ReactComponent as Calendar24 } from './calendar-24.svg';
import { ReactComponent as Info12 } from './info-12.svg';
import { ReactComponent as Info18 } from './info-18.svg';
import { ReactComponent as Info24 } from './info-24.svg';
import { ReactComponent as Help10 } from './help-fill-10.svg';
import { ReactComponent as Help16 } from './help-fill-16.svg';

export const systemIcons: { [key: string]: any } = {
    '18': {
        clear: BtnCross12,
        foldingArrow: FoldingArrow12,
        accept: Accept12,
        search: Search12,
        calendar: Calendar12,
        info: Info12,
        help: Help10,
    },
    '24': {
        clear: BtnCross12,
        foldingArrow: FoldingArrow12,
        accept: Accept12,
        search: Search12,
        calendar: Calendar12,
        info: Info12,
        help: Help10,
    },
    '30': {
        clear: BtnCross18,
        foldingArrow: FoldingArrow18,
        accept: Accept18,
        search: Search18,
        calendar: Calendar18,
        info: Info18,
        help: Help10,
    },
    '36': {
        clear: BtnCross18,
        foldingArrow: FoldingArrow18,
        accept: Accept18,
        search: Search18,
        calendar: Calendar18,
        info: Info18,
        help: Help16,
    },
    '42': {
        clear: BtnCross18,
        foldingArrow: FoldingArrow18,
        accept: Accept18,
        search: Search18,
        calendar: Calendar18,
        info: Info18,
        help: Help16,
    },
    '48': {
        clear: BtnCross24,
        foldingArrow: FoldingArrow24,
        accept: Accept24,
        search: Search24,
        calendar: Calendar24,
        info: Info24,
        help: Help16,
    },
    '60': {
        clear: BtnCross24,
        foldingArrow: FoldingArrow24,
        accept: Accept24,
        search: Search24,
        calendar: Calendar24,
        info: Info24,
        help: Help16,
    },
    none: {
        clear: BtnCross18,
        foldingArrow: FoldingArrow18,
        accept: Accept18,
        search: Search18,
        calendar: Calendar18,
        info: Info18,
        help: Help16,
    },
};
