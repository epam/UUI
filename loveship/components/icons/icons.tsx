import { ReactComponent as BtnCross12 } from './btn-cross-12.svg';
import { ReactComponent as BtnCross18 } from './btn-cross-18.svg';
import { ReactComponent as BtnCross24 } from './btn-cross-24.svg';
import { ReactComponent as FoldingArrow12 } from './folding-arrow-12.svg';
import { ReactComponent as FoldingArrow18 } from './folding-arrow-18.svg';
import { ReactComponent as FoldingArrow24 } from './folding-arrow-24.svg';
import { ReactComponent as Accept12 } from './accept-12.svg';
import { ReactComponent as Accept18 } from './accept-18.svg';
import { ReactComponent as Accept24 } from './accept-24.svg';
import { ReactComponent as Search12 } from './search-12.svg';
import { ReactComponent as Search18 } from './search-18.svg';
import { ReactComponent as Search24 } from './search-24.svg';
import { ReactComponent as Calendar12 } from './calendar-12.svg';
import { ReactComponent as Calendar18 } from './calendar-18.svg';
import { ReactComponent as Calendar24 } from './calendar-24.svg';
import { ReactComponent as Info12 } from './info-12.svg';
import { ReactComponent as Info18 } from './info-18.svg';
import { ReactComponent as Info24 } from './info-24.svg';

export const systemIcons: { [key: string]: any } = {
    '18': {
        clear: BtnCross12,
        foldingArrow: FoldingArrow12,
        accept: Accept12,
        search: Search12,
        calendar: Calendar12,
        info: Info12,
    },
    '24': {
        clear: BtnCross12,
        foldingArrow: FoldingArrow12,
        accept: Accept12,
        search: Search12,
        calendar: Calendar12,
        info: Info12,
    },
    '30': {
        clear: BtnCross12,
        foldingArrow: FoldingArrow12,
        accept: Accept12,
        search: Search12,
        calendar: Calendar12,
        info: Info12,
    },
    '36': {
        clear: BtnCross18,
        foldingArrow: FoldingArrow18,
        accept: Accept18,
        search: Search18,
        calendar: Calendar18,
        info: Info18,
    },
    '42': {
        clear: BtnCross18,
        foldingArrow: FoldingArrow18,
        accept: Accept18,
        search: Search18,
        calendar: Calendar18,
        info: Info18,
    },
    '48': {
        clear: BtnCross24,
        foldingArrow: FoldingArrow24,
        accept: Accept24,
        search: Search24,
        calendar: Calendar24,
        info: Info24,
    },
    '60': {
        clear: BtnCross24,
        foldingArrow: FoldingArrow24,
        accept: Accept24,
        search: Search24,
        calendar: Calendar24,
        info: Info24,
    },
    none: {
        clear: BtnCross18,
        foldingArrow: FoldingArrow18,
        accept: Accept18,
        search: Search18,
        calendar: Calendar18,
        info: Info18,
    },
};
