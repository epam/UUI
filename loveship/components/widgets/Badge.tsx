import { createSkinComponent, devLogger } from '@epam/uui-core';
import * as types from '../../components/types';
import * as uui from '@epam/uui';
import { EpamAdditionalColor, EpamPrimaryColor } from '../types';
import css from './Badge.module.scss';

const DEFAULT_SIZE = '18';

type BadgeColor = EpamPrimaryColor | EpamAdditionalColor | 'yellow'| 'orange' | 'purple' | 'cyan' | 'mint' | 'white'
| 'night100' | 'night300' | 'night600' | uui.BadgeProps['color'];

interface BadgeMods {
    /**
     * Defines component color.
     * @default 'sky'
     */
    color?: BadgeColor
    /**
     * Defines component shape style.
     * @default 'square'
     */
    shape?: types.ControlShape;
    /**
     * Defines component fill style.
     * @default 'solid'
     */
    fill?: uui.BadgeProps['fill'] | 'semitransparent';
    /**
     * Defines component size.
     * @default '18'
     */
    size?: uui.BadgeProps['size'] | '12';
}

function applyBadgeMods(mods: BadgeMods) {
    return [
        css['style-' + (mods.shape || 'square')],
        css[`fill-${mods.fill === 'semitransparent' ? 'outline' : (mods.fill || 'solid')}`],
        css['size-' + (mods.size || DEFAULT_SIZE)],
        css.root,
    ];
}

/** Represents the properties of a Badge component. */
export type BadgeProps = uui.BadgeCoreProps & BadgeMods;

export const Badge = /* @__PURE__ */createSkinComponent<uui.BadgeProps, BadgeProps>(
    uui.Badge as any, // TODO: remove it when BaseButton inheritance will be reworked
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<BadgeProps, 'fill'>({
                component: 'Badge',
                propName: 'fill',
                propValue: props.fill,
                propValueUseInstead: 'outline',
                condition: () => ['semitransparent'].indexOf(props.fill) !== -1,
            });
            devLogger.warnAboutDeprecatedPropValue<BadgeProps, 'size'>({ // TODO: remove size 12 from @uui/Badge: mapCountIndicatorSizes
                component: 'Badge',
                propName: 'size',
                propValue: props.size,
                propValueUseInstead: '18',
                condition: () => props.size === '12',
            });
            if (props.shape === 'square' || !props.shape) {
                devLogger.warn('(Badge) The "square" value of shape is deprecated and will be removed in future versions. Please use Tag component instead if you need square appearance or use "round" value.');
            }
        }
        return {
            color: props.color || 'sky',
            size: props.size || DEFAULT_SIZE,
            fill: props.fill === 'semitransparent' ? 'outline' : (props.fill || 'solid'),
        };
    },
    applyBadgeMods,
);
