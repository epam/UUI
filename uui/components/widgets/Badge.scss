@use  '../../assets/styles/index' as *;

.root {
    :global(.uui-count) {
        font-family: var(--font-semibold);
        margin: auto;
        height: 18px;
        line-height: 18px;
        font-size: 12px;
        border-radius: 9px;
        padding-left: 6px;
        padding-right: 6px;
        min-width: 6px;
        text-align: center;
    }

    :global(.uui-count):not(:first-child) {
        margin-left: 6px;
    }

    :global(.uui-caption) {
        font-family: var(--font-regular);
    }

    &:global(.uui-button-box) {
        border-width: 1px;

    }

    &.size-18 {
        min-width: 18px;
        min-height: 18px;
        padding-left: 3px;
        padding-right: 3px;
        border-radius: 9px;

        :global(.uui-count) {
            height: 12px;
            line-height: 12px;
            font-size: 8px;
            border-radius: 6px;
            padding-left: 3px;
            padding-right: 3px;
        }

        :global(.uui-count):not(:first-child) {
            margin-left: 3px;
        }

        :global(.uui-caption) {
            padding: 2px 3px;
            line-height: 12px;
            font-size: 12px;
        }

        &.fill-transparent {
            :global(.uui-caption) {
                margin-left: 8px;
            }

            &::before {
                width: 8px;
                height: 8px;
                top: 4px;
                left: 1px;
            }
        }
    }

    &.size-24 {
        @include button-horizontal-layout(24px, 1px);
        border-radius: 12px;
        min-height: 24px;

        :global(.uui-caption) {
            padding-top: 2px;
            padding-bottom: 2px;
            line-height: 18px;
            font-size: 14px;
        }

        &.fill-transparent {
            :global(.uui-caption) {
                margin-left: 12px;
            }

            &::before {
                width: 12px;
                height: 12px;
                top: 5px;
                left: 2px;
            }
        }
    }

    &.size-30 {
        @include button-horizontal-layout(30px, 1px);
        border-radius: 15px;
        min-height: 30px;

        :global(.uui-caption) {
            padding-top: 5px;
            padding-bottom: 5px;
            line-height: 18px;
            font-size: 14px;
        }

        &.fill-transparent {
            :global(.uui-caption) {
                margin-left: 12px;
            }

            &::before {
                width: 12px;
                height: 12px;
                top: 8px;
                left: 2px;
            }
        }
    }

    &.size-36 {
        @include button-horizontal-layout(36px, 1px);
        border-radius: 18px;
        min-height: 36px;

        :global(.uui-caption) {
            padding-top: 5px;
            padding-bottom: 5px;
            line-height: 24px;
            font-size: 14px;
        }

        &.fill-transparent {
            :global(.uui-caption) {
                margin-left: 12px;
            }

            &::before {
                width: 12px;
                height: 12px;
                top: 11px;
                left: 2px;
            }
        }
    }

    &.size-42 {
        @include button-horizontal-layout(42px, 1px);
        border-radius: 21px;
        min-height: 42px;

        :global(.uui-count) {
            height: 24px;
            line-height: 24px;
            font-size: 14px;
            border-radius: 12px;
            min-width: 12px;
        }

        :global(.uui-caption) {
            padding-top: 8px;
            padding-bottom: 8px;
            line-height: 24px;
            font-size: 16px;
        }

        &.fill-transparent {
            :global(.uui-caption) {
                margin-left: 12px;
            }

            &::before {
                width: 16px;
                height: 16px;
                top: 12px;
                left: 3px;
            }
        }
    }

    &.size-48 {
        @include button-horizontal-layout(48px, 1px);
        border-radius: 24px;
        min-height: 48px;

        :global(.uui-count) {
            height: 24px;
            line-height: 24px;
            font-size: 14px;
            border-radius: 12px;
            min-width: 12px;
        }

        :global(.uui-caption) {
            padding-top: 11px;
            padding-bottom: 11px;
            line-height: 24px;
            font-size: 16px;
        }

        &.fill-transparent {
            :global(.uui-caption) {
                margin-left: 12px;
            }

            &::before {
                width: 16px;
                height: 16px;
                top: 15px;
                left: 3px;
            }
        }
    }

    &.fill-solid {
        background-color: var(--uui-badge-bg);
        color: var(--uui-badge-caption-solid);
        fill: var(--uui-badge-caption-solid);
        border-color: var(--uui-badge-bg);

        :global(.uui-count) {
            background-color: var(--uui-badge-count-bg);
            color: var(--uui-badge-count);
        }

        &:global(.-clickable) {
            &:hover {
                background-color: var(--uui-badge-bg-hover);
                border-color: var(--uui-badge-bg-hover);
                cursor: pointer;
            }
        }
    }

    //&.fill-semitransparent {
    //    background-color: var(--bg-color-semitransparent);
    //    color: var(--uui-caption-semitransparent);
    //    fill: var(--uui-caption-semitransparent);
    //    border-color: var(--border-color-semitransparent);
    //
    //    :global(.uui-count) {
    //        background-color: var(--count-bg-color);
    //        color: var(--count-color);
    //    }
    //
    //    &:global(.-clickable) {
    //        &:hover {
    //            background-color: var(--border-color-semitransparent);
    //            color: var(--uui-caption-semitransparent-hover);
    //            fill: var(--uui-caption-semitransparent-hover);
    //            cursor: pointer;
    //        }
    //    }
    //}
    //
    //&.fill-transparent {
    //    background-color: transparent;
    //    color: var(--uui-caption-transparent);
    //    fill: var(--bg-color);
    //    border-color: transparent;
    //
    //    &:global(.-clickable) {
    //        &:hover {
    //            fill: var(--bg-color-hover);
    //            cursor: pointer;
    //        }
    //    }
    //
    //    &::before {
    //        position: absolute;
    //        content: '';
    //        border-radius: 50%;
    //        background-color: var(--bg-color);
    //    }
    //}
}
