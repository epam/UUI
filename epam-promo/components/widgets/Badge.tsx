import * as React from 'react';
import { withMods } from '@epam/uui-core';
import { Button, ButtonProps } from '@epam/uui-components';
import { EpamAdditionalColor } from '../types';
import { systemIcons } from '../../icons/icons';
// import buttonCss from '../buttons/Button.scss';
import styles from '../../assets/styles/colorvars/widgets/badge-colorvars.scss';
import css from './Badge.scss';

const defaultSize = '36';

const mapSize = {
    '48': '48',
    '42': '48',
    '36': '36',
    '30': '30',
    '24': '30',
    '18': '18',
};

export interface BadgeMods {
    fill?: 'solid' | 'semitransparent' | 'transparent';
    size?: '18' | '24' | '30' | '36' | '42' | '48';
    color?: EpamAdditionalColor | 'gray30';
}

export function applyBadgeMods(mods: BadgeMods) {
    return [
        // buttonCss.root,
        css['size-' + (mods.size || defaultSize)],
        css['fill-' + (mods.fill || 'solid')],
        styles['badge-color-' + (mods.color || 'blue')],
        css.root,
    ];
}

export const Badge = withMods<ButtonProps, BadgeMods>(
    Button,
    applyBadgeMods,
    (props) => ({
        dropdownIcon: systemIcons[props.size && mapSize[props.size] || defaultSize].foldingArrow,
        clearIcon: systemIcons[props.size && mapSize[props.size] || defaultSize].clear,
        countPosition: 'left',
    }),
);
