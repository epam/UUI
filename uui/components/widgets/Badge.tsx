import * as React from 'react';
import { withMods } from '@epam/uui-core';
import { Button, ButtonProps } from '@epam/uui-components';
import { systemIcons } from '../../icons/icons';
import buttonCss from '../buttons/Button/Button.scss';
import css from './Badge.scss';
import '../../assets/styles/variables/widgets/badge.scss';

const defaultSize = '36';

const mapSize = {
    '48': '48',
    '42': '48',
    '36': '36',
    '30': '30',
    '24': '30',
    '18': '18',
};

export type BadgeColor = 'info' | 'success' | 'warning' | 'error';
export type BadgeFill = 'solid';
export type BadgeSize = '18' | '24' | '30' | '36' | '42' | '48';

export interface BadgeMods {
    fill?: BadgeFill;
    size?: BadgeSize;
    color?: BadgeColor;
}

export interface BadgeProps extends ButtonProps, BadgeMods {}

export function applyBadgeMods(mods: BadgeMods) {
    return [

        `badge-${(mods.color || 'info')}`,
        buttonCss.root,
        css['size-' + (mods.size || defaultSize)],
        css['fill-' + (mods.fill || 'solid')],
        css.root,
    ];
}

export const Badge = withMods<ButtonProps, BadgeMods>(
    Button,
    applyBadgeMods,
    (props) => ({
        dropdownIcon: systemIcons[props.size && mapSize[props.size] || defaultSize].foldingArrow,
        clearIcon: systemIcons[props.size && mapSize[props.size] || defaultSize].clear,
        countPosition: 'left',
    }),
);
