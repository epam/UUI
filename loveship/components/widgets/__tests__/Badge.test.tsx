import React from 'react';
import renderer from 'react-test-renderer';
import { Badge } from '../Badge';
import { systemIcons } from '../../icons/icons';
const acceptIcon = systemIcons[18].accept;

describe('Badge', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<Badge />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer
            .create(
                <Badge
                    shape="square"
                    fill="light"
                    size="12"
                    caption="Test badge"
                    icon={acceptIcon}
                    count={12}
                    onIconClick={jest.fn()}
                    onClick={jest.fn()}
                    onClear={jest.fn()}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
