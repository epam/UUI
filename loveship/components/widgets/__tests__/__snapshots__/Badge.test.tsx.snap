// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should be rendered correctly 1`] = `
<button
  className="container uui-button-box uui-enabled root style-square size-18 fill-solid color-sky root"
  href={null}
  onClick={[Function]}
  role="button"
  tabIndex={-1}
  type="button"
/>
`;

exports[`Badge should be rendered correctly with props 1`] = `
<button
  className="container uui-button-box uui-enabled -clickable root style-square size-12 fill-light color-sky root"
  href={null}
  onClick={[Function]}
  role="button"
  tabIndex={0}
  type="button"
>
  <div
    className="container uui-icon uui-enabled -clickable"
    onClick={[MockFunction]}
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
  <div
    className="uui-count"
  >
    12
  </div>
  <div
    className="uui-caption"
  >
    Test badge
  </div>
  <div
    className="container uui-icon uui-enabled -clickable -clickable"
    onClick={[MockFunction]}
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
</button>
`;
